subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/WatchRootUtil.java

```diff
@@ -79,6 +79,9 @@
             default:
                 throw new AssertionError();
         }
+        if (ancestorToWatch == null) {
+            return ImmutableList.of(path);
+        }
```

Comment:
We should have a comment here explaining that this is about watching the root of a file-system.

Reply:
Does this mean that if we have a typo in a file's path (e.g. like `C:\rogram Files\Something\Something`), we'll end up watching the whole drive? I think we need to be more explicit, and only check if the direct parent of the actual path is `null`.

Reply:
We already do watch the root drive if there is a type in the file's path. This PR doesn't change anything about that. `findFirstExistingAncestor` also never returns `null` and fails when the drive is missing. I wonder if we can do something different than failing when the whole drive is missing on `Windows`. I suppose we would need to find a way to listen for drive creation events.

Reply:
I'm not so worried about drive creation events, at least not until we have a concrete use case for where it's broken.

I'm a bit more worried about missing files potentially leading to watching the entire drive. What about this scenario:

* I'm trying to watch `C:\a\b\c\d\e`, and `C:\a` (along with the rest) does not exist, so I end up watching `C:\`
* Then `C:\a\b\c` gets created. In this case it seems we should stop watching `C:\`, and start watching `C:\a\b\c` instead. I don't think we do this right now, right?

Reply:
We already do this (watch the subdirectory when it appears). I added a test.